File: Programmers/Lv2/Lv2_큰수만들기.py
def solution(number, k):
    stack = [number[0]]
    for num in number[1:]:
        while len(stack) > 0 and stack[-1] < num and k > 0:
            k -= 1
            stack.pop()
        stack.append(num)
    if k != 0:                  # 紐⑤몢 ?묎컳? ?섎줈 ?ㅼ뼱??寃쎌슦 ?↔린
        stack = stack[:-k]      # ?ㅼ뿉?쒕???k媛??쒓굅?섍린
    return ''.join(stack)

print(solution("111111", 3))

File: Programmers/Lv2/Lv2_큰수만들기.cpp
#include <string>
#include <vector>
using namespace std;

string solution(string number, int k) {
	string answer = "";
	int size = number.length() - k;
	int size2 = size;
	int len = k;
	int end = size;

	while (answer.length()!=size2) {
		int high = -1;
		int index = -1;
		char temp = '0';

		int end = number.length() - size + 1;
		if (number.length() >= end) {
			for (int i = 0; i < end; i++) {
				if (high < number[i] - 48) {
					high = number[i] - 48;
					index = i;
				}
			}
			answer.push_back(temp + high);
			number = number.substr(index + 1);
			size--;
		}

		if (number.length() == size) {
			for (int i = 0; i < number.length(); i++)
				answer.push_back(number[i]);
		}
	}
	return answer;
}

